@thockin It breaks for me the same way locally:
```
$ make release KUBE_RELEASE_RUN_TESTS=n
+++ [0405 23:22:52] Verifying Prerequisites....
+++ [0405 23:22:53] Building Docker image kube-build:build-fd5ab429b0-5-v1.30.0-go1.22.1-bullseye.0
+++ [0405 23:22:56] Syncing sources to container
+++ [0405 23:22:58] Running build command...
+++ [0405 23:23:08] Multiple platforms requested and available 54G >= threshold 20G, building platforms in parallel
+++ [0405 23:23:08] Building go targets for {linux/amd64 linux/arm64 linux/s390x linux/ppc64le} in parallel (output will appear in a burst when complete):
    cluster/gce/gci/mounter
    cmd/kube-apiserver
    cmd/kube-controller-manager
    cmd/kube-proxy
    cmd/kube-scheduler
    cmd/kubeadm
    cmd/kubelet
    staging/src/k8s.io/apiextensions-apiserver
    staging/src/k8s.io/component-base/logs/kube-log-runner
    staging/src/k8s.io/kube-aggregator
+++ [0405 23:23:08] linux/amd64: build started
    k8s.io/apiextensions-apiserver (static)
    k8s.io/component-base/logs/kube-log-runner (static)
    k8s.io/kube-aggregator (static)
    k8s.io/kubernetes/cluster/gce/gci/mounter (static)
    k8s.io/kubernetes/cmd/kube-apiserver (static)
    k8s.io/kubernetes/cmd/kube-controller-manager (static)
    k8s.io/kubernetes/cmd/kube-proxy (static)
    k8s.io/kubernetes/cmd/kube-scheduler (static)
    k8s.io/kubernetes/cmd/kubeadm (static)
    k8s.io/kubernetes/cmd/kubelet (non-static)
+++ [0405 23:23:20] linux/amd64: build finished
+++ [0405 23:23:08] linux/arm64: build started
    k8s.io/apiextensions-apiserver (static)
    k8s.io/component-base/logs/kube-log-runner (static)
    k8s.io/kube-aggregator (static)
    k8s.io/kubernetes/cluster/gce/gci/mounter (static)
    k8s.io/kubernetes/cmd/kube-apiserver (static)
    k8s.io/kubernetes/cmd/kube-controller-manager (static)
    k8s.io/kubernetes/cmd/kube-proxy (static)
    k8s.io/kubernetes/cmd/kube-scheduler (static)
    k8s.io/kubernetes/cmd/kubeadm (static)
    k8s.io/kubernetes/cmd/kubelet (non-static)
+++ [0405 23:23:21] linux/arm64: build finished
+++ [0405 23:23:08] linux/s390x: build started
    k8s.io/apiextensions-apiserver (static)
    k8s.io/component-base/logs/kube-log-runner (static)
    k8s.io/kube-aggregator (static)
    k8s.io/kubernetes/cluster/gce/gci/mounter (static)
    k8s.io/kubernetes/cmd/kube-apiserver (static)
    k8s.io/kubernetes/cmd/kube-controller-manager (static)
    k8s.io/kubernetes/cmd/kube-proxy (static)
    k8s.io/kubernetes/cmd/kube-scheduler (static)
    k8s.io/kubernetes/cmd/kubeadm (static)
    k8s.io/kubernetes/cmd/kubelet (non-static)
+++ [0405 23:23:20] linux/s390x: build finished
+++ [0405 23:23:08] linux/ppc64le: build started
    k8s.io/apiextensions-apiserver (static)
    k8s.io/component-base/logs/kube-log-runner (static)
    k8s.io/kube-aggregator (static)
    k8s.io/kubernetes/cluster/gce/gci/mounter (static)
    k8s.io/kubernetes/cmd/kube-apiserver (static)
    k8s.io/kubernetes/cmd/kube-controller-manager (static)
    k8s.io/kubernetes/cmd/kube-proxy (static)
    k8s.io/kubernetes/cmd/kube-scheduler (static)
    k8s.io/kubernetes/cmd/kubeadm (static)
    k8s.io/kubernetes/cmd/kubelet (non-static)
+++ [0405 23:23:21] linux/ppc64le: build finished
+++ [0405 23:23:23] Multiple platforms requested and available 55G >= threshold 20G, building platforms in parallel
+++ [0405 23:23:23] Building go targets for {linux/amd64 linux/arm64 linux/s390x linux/ppc64le windows/amd64} in parallel (output will appear in a burst when complete):
    cmd/kube-proxy
    cmd/kubeadm
    cmd/kubelet
    staging/src/k8s.io/component-base/logs/kube-log-runner
+++ [0405 23:23:23] linux/amd64: build started
    k8s.io/component-base/logs/kube-log-runner (static)
    k8s.io/kubernetes/cmd/kube-proxy (static)
    k8s.io/kubernetes/cmd/kubeadm (static)
    k8s.io/kubernetes/cmd/kubelet (non-static)
+++ [0405 23:23:30] linux/amd64: build finished
+++ [0405 23:23:23] linux/arm64: build started
    k8s.io/component-base/logs/kube-log-runner (static)
    k8s.io/kubernetes/cmd/kube-proxy (static)
    k8s.io/kubernetes/cmd/kubeadm (static)
    k8s.io/kubernetes/cmd/kubelet (non-static)
+++ [0405 23:23:31] linux/arm64: build finished
+++ [0405 23:23:23] linux/s390x: build started
    k8s.io/component-base/logs/kube-log-runner (static)
    k8s.io/kubernetes/cmd/kube-proxy (static)
    k8s.io/kubernetes/cmd/kubeadm (static)
    k8s.io/kubernetes/cmd/kubelet (non-static)
+++ [0405 23:23:30] linux/s390x: build finished
+++ [0405 23:23:23] linux/ppc64le: build started
    k8s.io/component-base/logs/kube-log-runner (static)
    k8s.io/kubernetes/cmd/kube-proxy (static)
    k8s.io/kubernetes/cmd/kubeadm (static)
    k8s.io/kubernetes/cmd/kubelet (non-static)
+++ [0405 23:23:32] linux/ppc64le: build finished
+++ [0405 23:23:23] windows/amd64: build started
    k8s.io/component-base/logs/kube-log-runner (static)
    k8s.io/kubernetes/cmd/kube-proxy (static)
    k8s.io/kubernetes/cmd/kubeadm (static)
    k8s.io/kubernetes/cmd/kubelet (non-static)
+++ [0405 23:23:32] windows/amd64: build finished
+++ [0405 23:23:33] Multiple platforms requested and available 54G >= threshold 20G, building platforms in parallel
+++ [0405 23:23:33] Building go targets for {linux/amd64 linux/386 linux/arm linux/arm64 linux/s390x linux/ppc64le darwin/amd64 darwin/arm64 windows/amd64 windows/386 windows/arm64} in parallel (output will appear in a burst when complete):
    cmd/kubectl
    cmd/kubectl-convert
+++ [0405 23:23:33] linux/amd64: build started
    k8s.io/kubernetes/cmd/kubectl (static)
    k8s.io/kubernetes/cmd/kubectl-convert (static)
+++ [0405 23:23:38] linux/amd64: build finished
+++ [0405 23:23:33] linux/386: build started
    k8s.io/kubernetes/cmd/kubectl (static)
    k8s.io/kubernetes/cmd/kubectl-convert (static)
+++ [0405 23:23:41] linux/386: build finished
+++ [0405 23:23:33] linux/arm: build started
    k8s.io/kubernetes/cmd/kubectl (static)
    k8s.io/kubernetes/cmd/kubectl-convert (static)
+++ [0405 23:23:41] linux/arm: build finished
+++ [0405 23:23:33] linux/arm64: build started
    k8s.io/kubernetes/cmd/kubectl (static)
    k8s.io/kubernetes/cmd/kubectl-convert (static)
+++ [0405 23:23:38] linux/arm64: build finished
+++ [0405 23:23:33] linux/s390x: build started
    k8s.io/kubernetes/cmd/kubectl (static)
    k8s.io/kubernetes/cmd/kubectl-convert (static)
+++ [0405 23:23:38] linux/s390x: build finished
+++ [0405 23:23:33] linux/ppc64le: build started
    k8s.io/kubernetes/cmd/kubectl (static)
    k8s.io/kubernetes/cmd/kubectl-convert (static)
+++ [0405 23:23:38] linux/ppc64le: build finished
+++ [0405 23:23:33] darwin/amd64: build started
    k8s.io/kubernetes/cmd/kubectl (static)
    k8s.io/kubernetes/cmd/kubectl-convert (static)
+++ [0405 23:23:41] darwin/amd64: build finished
+++ [0405 23:23:33] darwin/arm64: build started
    k8s.io/kubernetes/cmd/kubectl (static)
    k8s.io/kubernetes/cmd/kubectl-convert (static)
+++ [0405 23:23:41] darwin/arm64: build finished
+++ [0405 23:23:33] windows/amd64: build started
    k8s.io/kubernetes/cmd/kubectl (static)
    k8s.io/kubernetes/cmd/kubectl-convert (static)
+++ [0405 23:23:38] windows/amd64: build finished
+++ [0405 23:23:33] windows/386: build started
    k8s.io/kubernetes/cmd/kubectl (static)
    k8s.io/kubernetes/cmd/kubectl-convert (static)
+++ [0405 23:23:41] windows/386: build finished
+++ [0405 23:23:33] windows/arm64: build started
    k8s.io/kubernetes/cmd/kubectl (static)
    k8s.io/kubernetes/cmd/kubectl-convert (static)
+++ [0405 23:23:41] windows/arm64: build finished
+++ [0405 23:23:42] Multiple platforms requested and available 54G >= threshold 20G, building platforms in parallel
+++ [0405 23:23:42] Building go targets for {linux/amd64 linux/arm64 linux/s390x linux/ppc64le darwin/amd64 darwin/arm64 windows/amd64 windows/arm64} in parallel (output will appear in a burst when complete):
    ginkgo
    test/conformance/image/go-runner
    test/e2e/e2e.test
+++ [0405 23:23:42] linux/amd64: build started
    github.com/onsi/ginkgo/v2/ginkgo (non-static)
    k8s.io/kubernetes/test/conformance/image/go-runner (non-static)
    k8s.io/kubernetes/test/e2e/e2e.test (test)
+++ [0405 23:23:53] linux/amd64: build finished
+++ [0405 23:23:42] linux/arm64: build started
    github.com/onsi/ginkgo/v2/ginkgo (non-static)
    k8s.io/kubernetes/test/conformance/image/go-runner (non-static)
    k8s.io/kubernetes/test/e2e/e2e.test (test)
+++ [0405 23:23:52] linux/arm64: build finished
+++ [0405 23:23:42] linux/s390x: build started
    github.com/onsi/ginkgo/v2/ginkgo (non-static)
    k8s.io/kubernetes/test/conformance/image/go-runner (non-static)
    k8s.io/kubernetes/test/e2e/e2e.test (test)
+++ [0405 23:23:52] linux/s390x: build finished
+++ [0405 23:23:42] linux/ppc64le: build started
    github.com/onsi/ginkgo/v2/ginkgo (non-static)
    k8s.io/kubernetes/test/conformance/image/go-runner (non-static)
    k8s.io/kubernetes/test/e2e/e2e.test (test)
+++ [0405 23:23:54] linux/ppc64le: build finished
+++ [0405 23:23:42] darwin/amd64: build started
    github.com/onsi/ginkgo/v2/ginkgo (non-static)
    k8s.io/kubernetes/test/conformance/image/go-runner (non-static)
    k8s.io/kubernetes/test/e2e/e2e.test (test)
+++ [0405 23:23:50] darwin/amd64: build finished
+++ [0405 23:23:42] darwin/arm64: build started
    github.com/onsi/ginkgo/v2/ginkgo (non-static)
    k8s.io/kubernetes/test/conformance/image/go-runner (non-static)
    k8s.io/kubernetes/test/e2e/e2e.test (test)
+++ [0405 23:23:51] darwin/arm64: build finished
+++ [0405 23:23:42] windows/amd64: build started
    github.com/onsi/ginkgo/v2/ginkgo (non-static)
    k8s.io/kubernetes/test/conformance/image/go-runner (non-static)
    k8s.io/kubernetes/test/e2e/e2e.test (test)
+++ [0405 23:23:49] windows/amd64: build finished
+++ [0405 23:23:42] windows/arm64: build started
    github.com/onsi/ginkgo/v2/ginkgo (non-static)
    k8s.io/kubernetes/test/conformance/image/go-runner (non-static)
    k8s.io/kubernetes/test/e2e/e2e.test (test)
+++ [0405 23:23:50] windows/arm64: build finished
+++ [0405 23:23:55] Multiple platforms requested and available 55G >= threshold 20G, building platforms in parallel
+++ [0405 23:23:55] Building go targets for {linux/amd64 linux/arm64 linux/s390x linux/ppc64le} in parallel (output will appear in a burst when complete):
    cmd/kubemark
    ginkgo
    test/e2e_node/e2e_node.test
+++ [0405 23:23:55] linux/amd64: build started
    github.com/onsi/ginkgo/v2/ginkgo (non-static)
    k8s.io/kubernetes/cmd/kubemark (static)
    k8s.io/kubernetes/test/e2e_node/e2e_node.test (test)
+++ [0405 23:24:06] linux/amd64: build finished
+++ [0405 23:23:55] linux/arm64: build started
    github.com/onsi/ginkgo/v2/ginkgo (non-static)
    k8s.io/kubernetes/cmd/kubemark (static)
    k8s.io/kubernetes/test/e2e_node/e2e_node.test (test)
+++ [0405 23:24:07] linux/arm64: build finished
+++ [0405 23:23:55] linux/s390x: build started
    github.com/onsi/ginkgo/v2/ginkgo (non-static)
    k8s.io/kubernetes/cmd/kubemark (static)
    k8s.io/kubernetes/test/e2e_node/e2e_node.test (test)
+++ [0405 23:24:06] linux/s390x: build finished
+++ [0405 23:23:55] linux/ppc64le: build started
    github.com/onsi/ginkgo/v2/ginkgo (non-static)
    k8s.io/kubernetes/cmd/kubemark (static)
    k8s.io/kubernetes/test/e2e_node/e2e_node.test (test)
+++ [0405 23:24:08] linux/ppc64le: build finished
+++ [0405 23:24:14] Syncing out of container
receiving file list ... done

sent 192 bytes  received 197.603 bytes  131.863,33 bytes/sec
total size is 29.700.760  speedup is 150,16
+++ [0405 23:24:16] Starting tarball: client *-*
+++ [0405 23:24:16] Building tarball: src
+++ [0405 23:24:16] Building tarball: manifests
+++ [0405 23:24:16] Waiting on tarballs
tar: Removing leading `/' from member names
cp: cannot stat '/home/ed/go/src/k8s.io/kubernetes/_output/dockerized/bin/*/*/kubectl': No such file or directory
cp: cannot stat '/home/ed/go/src/k8s.io/kubernetes/_output/dockerized/bin/*/*/kubectl-convert': No such file or directory
tar: Removing leading `/' from hard link targets
!!! [0405 23:24:16] Call tree:
!!! [0405 23:24:16]  1: /home/ed/go/src/k8s.io/kubernetes/build/lib/release.sh:48 kube::release::package_client_tarballs(...)
!!! [0405 23:24:16]  2: build/release.sh:45 kube::release::package_tarballs(...)
!!! [0405 23:24:16] client tarball creation failed
tar: Removing leading `/' from member names
tar: Removing leading `/' from hard link targets
tar: Removing leading `/' from member names
tar: Removing leading `/' from member names
tar: Removing leading `/' from hard link targets
tar: Removing leading `/' from member names
^Cmake: *** [Makefile:376: release] Interrupt

```

--- a/hack/lib/golang.sh
+++ b/hack/lib/golang.sh
@@ -961,7 +961,7 @@ kube::golang::build_binaries() {
       cat "/tmp//${platform//\//_}.build"
     done
 
-    exit "${fails}"
+    return "${fails}"
   else
     for platform in "${platforms[@]}"; do
       kube::log::status "Building go targets for ${platform}"

--- a/hack/lib/golang.sh
+++ b/hack/lib/golang.sh
@@ -635,6 +635,7 @@ kube::golang::place_bins() {
       ln -s "${KUBE_OUTPUT_BIN}/${platform}" "${THIS_PLATFORM_BIN}"
     fi
 
+    V=3 kube::log::status "Placing binaries for ${platform} in ${KUBE_OUTPUT_BIN}/${platform}"
     local full_binpath_src="${KUBE_GOPATH}/bin${platform_src}"
     if [[ -d "${full_binpath_src}" ]]; then
       mkdir -p "${KUBE_OUTPUT_BIN}/${platform}"
@@ -795,7 +796,7 @@ kube::golang::build_binaries_for_platform() {
     fi
    done
 
-  V=2 kube::log::info "Env for ${platform}: GOOS=${GOOS-} GOARCH=${GOARCH-} GOROOT=${GOROOT-} CGO_ENABLED=${CGO_ENABLED-} CC=${CC-}"
+  V=2 kube::log::info "Env for ${platform}: GOPATH=${GOPATH-} GOOS=${GOOS-} GOARCH=${GOARCH-} GOROOT=${GOROOT-} CGO_ENABLED=${CGO_ENABLED-} CC=${CC-}"
   V=3 kube::log::info "Building binaries with GCFLAGS=${gogcflags} LDFLAGS=${goldflags}"
 
   local -a build_args

